### PR TITLE
tools/stellar-archivist: support custom S3 endpoints

### PR DESCRIPTION
**Pull request by nebolsin**
This PR adds custom S3 endpoint support to `stellar-archivist` tool, so that it can be used with any S3-compatible storage (like [DigitalOcean Spaces](https://www.digitalocean.com/products/spaces/)).

### CODE REVIEW

File: tools/stellar-archivist/internal/s3_archive.go [hunk 1]
```diff
@@ -101,15 +101,21 @@ func (b *S3ArchiveBackend) CanListFiles() bool {
 	return true
 }
 
-func MakeS3Backend(bucket string, prefix string, opts *ConnectOptions) ArchiveBackend {
-	cfg := aws.Config{}
-	if opts != nil && opts.S3Region != "" {
```
Comment (nullstyle) [inline tools/stellar-archivist/internal/s3_archive.go]: Please do not remove the nil-check here.  Either re-type `opts` as `ConnectOptions` or keep the check in place.

Comment (nebolsin) [reply tools/stellar-archivist/internal/s3_archive.go]: Good catch. Re-typed to pass `ConnectOptions` as values everywhere.

**Comment by nebolsin** [comment]
@nullstyle Thanks for reviewing. I addressed your comment and rebased against current master.

**Comment by nullstyle** [comment]
LGTM!  Thanks so much for the contribution @nebolsin

**Pull request merged**